MozillaFoundation/foundation.mozilla.org
# Bugfix/tp1 249 profile blocks respect numbers

# Description

The purpose of this fix was to make sure Profile blocks worked properly together on the same page regarding the field for number of results. The blocks revised are: Profile By Id, Profile Listing, Profile Directory. The issue mentioned a conflict with "profile by id" block and "profile listing" in the number of results showed when "profile directory" was added on the same page, not respecting the max number of profiles previously set. However, it seems that this issue was partially fixed, validating now that "Max number of results" is correctly taken for profile by id and profile listing when profile directory on the same page. Nevertheless, number od results was not working correctly for profile directory, so `no_limit` argument was updated to take correctlty the value. It Is important to mention that even when this "Max number of results" is set on the CMS, the profile directory block filters by "year", making this field only work on the initial page display before to change from year to year. 

# How to test
- Login into the CMS
- Go to home pages and create a "test" default page.
- On the body field search for "profile" and add: Profile  by id, Profile listing, and Profile directory.
![image](https://github.com/user-attachments/assets/295ba531-30cf-49b8-a568-d14b92c1e768)

- In the Profile by ID block add any number of IDs (Eg. 20, 30, 40, 50)
- In Profile listing block on the "Max number of results" field add the number 4 (can be any number > 1).
- In Profile directory block on the "Max number of results" field add the number 4 (can be any number > 1).
- Save the draft and publish the page.

You will see profile directory with the number of profiles specified:
![image](https://github.com/user-attachments/assets/ea67207c-3d58-4820-98f3-1db3773bdae1)

In the Profile listing section you will find the correct amount of profiles listed as specified:
![image](https://github.com/user-attachments/assets/9b0c76e0-3a19-490c-8280-b1d2950c5e87)

Profile by ID section will have as well the amount of profiles added in the ID field.
![image](https://github.com/user-attachments/assets/079289f9-ead4-4c0e-bd27-1bab8587fbfb)


Link to sample test page: https://foundation-s-bugfix-tp1-ygoeuo.herokuapp.com/en/
Related PRs/issues: https://github.com/MozillaFoundation/foundation.mozilla.org/issues/7580

# Checklist



**Tests**

**Changes in Models:**

**Documentation:**

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**
